sds/src/scanner/regex_rule/regex_cache_store.rs

```diff
@@ -0,0 +1,45 @@
+use crate::scanner::regex_rule::regex_store::{RegexCacheKey, SharedRegex};
+use crate::SharedPool;
+use lazy_static::lazy_static;
+use regex_automata::meta::Regex as MetaRegex;
+use slotmap::SecondaryMap;
+use std::sync::Arc;
+
+lazy_static! {
+    static ref REGEX_CACHE_STORE: Arc<SharedPool<Box<RegexCaches>>> =
+        Arc::new(SharedPool::new(|| Box::new(RegexCaches::new())));
+}
+
+pub fn access_regex_caches<T>(func: impl FnOnce(&mut RegexCaches) -> T) -> T {
+    // This function isn't strictly necessary, but it makes it easier to change the implementation
+    // later
+    let mut caches = REGEX_CACHE_STORE.get();
+    func(&mut caches)
+}
+
+pub struct RegexCaches {
+    map: SecondaryMap<RegexCacheKey, regex_automata::meta::Cache>,
```

Comment:
for my own understanding I don't get the main difference between secondaryMap and AHashMap what the benefit of SecondaryMap here ?

Reply:
You can read more about it here: https://docs.rs/slotmap/latest/slotmap/secondary/struct.SecondaryMap.html

The difference is quite important and it comes down to the implementation. Internally it's actually just wrapping a `Vec`, so lookups are faster (no hashing required). The main benefit is due to how the keys are managed. Once the primary `SlotMap` has a key removed, new keys can re-use the same index in the `Vec` so the `SecondaryMap` will automatically clean up memory from unused cached without having to manually remove old entries from every `SecondaryMap` manually.

Reply:
ok I was missing the SlotMap part but it now makes sense thks